In [ ]:
import pandas as pd

with open("../data/secchi_match_Zsd_0D_9-9Valid_SatRangeLessThan3_RatioBelow2_Secchi_Zsd_LinLin_selected.csv") as f:
    header = f.readline().strip().split(",")
    rows = [l.strip().split(",") for l in f.readlines()]

df = pd.DataFrame([r[:47] for r in rows], columns=header)
df.to_parquet("../data/Parquet/satellite.parquet", index=False)
print(f"Saved satellite.parquet — {len(df)} rows, {len(df.columns)} columns")

Saved satellite.parquet — 443 rows, 47 columns


In [13]:
import pandas as pd

secchi = pd.read_parquet("../data/Parquet/New_estchlsummed.parquet")
satellite = pd.read_parquet("../data/Parquet/satellite.parquet")

secchi_ids = set(secchi["Cast_ID"].dropna())
satellite_ids = set(satellite["Cast_ID"].dropna())

matching = secchi_ids & satellite_ids

print(f"secchi    — unique Cast_IDs : {len(secchi_ids)}")
print(f"satellite — unique Cast_IDs : {len(satellite_ids)}")
print(f"Exact matches               : {len(matching)}")

if matching:
    print(f"\nSample matching IDs:")
    for i in list(matching)[:5]:
        print(f"  {i!r}")

secchi    — unique Cast_IDs : 1971
satellite — unique Cast_IDs : 443
Exact matches               : 274

Sample matching IDs:
  '20-1011NM-PR-314-1738-08000700'
  '20-1011NM-MX-303-2230-09330800'
  '20-0701JD-PR-016-1729-09001200'
  '20-1607OS-PR-203-1609-08180469'
  '20-0604NM-PR-095-1923-09001100'


There are 2 Cast ID columns in Secchi (New_estchlsummed.parquet): "CAST_ID" from the CTD data and "Cast_ID" from the original Secchi data (since this dataset was initially 2 separate datasets merged together).

So, by using the "Cast_ID" column in Secchi to merge with the satellite dataset, the names line up and there are 274 matches. 

Question: if Pointvalue in satellite is the Kd490 estimated Secchi depth, what is the Secchi column in that dataset? The values seem similar but not exact, different distributions.

In [ ]:
import pandas as pd

secchi = pd.read_parquet("../data/Parquet/New_estchlsummed.parquet")
satellite = pd.read_parquet("../data/Parquet/satellite.parquet")

merged = secchi.merge(satellite, on="Cast_ID", how="inner")

merged.to_parquet("../data/Parquet/satellite_merged.parquet", index=False)
print(f"Saved satellite_merged.parquet — {len(merged)} rows")

Saved satellite_merged.parquet — 274 rows
